src/librustc_lint/builtin.rs

```diff
@@ -1180,6 +1180,24 @@ impl LintPass for InvalidNoMangleItems {
 
 impl<'a, 'tcx> LateLintPass<'a, 'tcx> for InvalidNoMangleItems {
     fn check_item(&mut self, cx: &LateContext, it: &hir::Item) {
+        let suggest_make_pub = |vis: &hir::Visibility, err: &mut DiagnosticBuilder| {
+            let suggestion = match &vis {
+                hir::Visibility::Inherited => {
+                    // insert `pub` plus a space at the beginning of the item
+                    Some((it.span.shrink_to_lo(), "pub ".to_owned()))
+                },
+                hir::Visibility::Restricted { span, .. } |
+                hir::Visibility::Crate(span, _) => {
+                    // replace the existing visibility modifier
+                    Some((span.clone(), "pub".to_owned()))
+                },
+                hir::Visibility::Public(_) => None
+            };
+            if let Some((replace_span, replacement)) = suggestion {
+                err.span_suggestion(replace_span, "try making it public", replacement);
```

Comment:
Mark this MaybeIncorrect?

(such changes likely need review, hence it's not MachineApplicable)